python/ray/_private/internal_api.py

```diff
@@ -30,20 +39,18 @@ def memory_summary(
 ):
     from ray.dashboard.memory_utils import memory_summary
 
-    address = services.canonicalize_bootstrap_address_or_die(address)
+    state = get_state_from_address(address)
+    reply = get_memory_info_reply(state)
 
-    state = GlobalState()
-    options = GcsClientOptions.from_gcs_address(address)
-    state._initialize_global_state(options)
     if stats_only:
-        return get_store_stats(state)
+        return store_stats_summary(reply)
     return memory_summary(
         state, group_by, sort_by, line_wrap, units, num_entries
-    ) + get_store_stats(state)
+    ) + store_stats_summary(reply)
 
 
-def get_store_stats(state, node_manager_address=None, node_manager_port=None):
-    """Returns a formatted string describing memory usage in the cluster."""
+def get_memory_info_reply(state, node_manager_address=None, node_manager_port=None):
```

Comment:
For my own understanding, where do I find the definition of the reply object?

Reply:
I think its `ObjectStoreStats` in `node_manager.proto`